proposals/0000-type-vs-constraint.rst

```diff
@@ -0,0 +1,365 @@
+Separating Type from Constraint
```

Comment:
One more comment to object to the title. This proposal doesn't separate `Type` from `Constraint`: they are already separate. One of the motivations is that Core treats `Type` and `Constraint` as separate, but Core is out of scope for GHC proposal. At any rate, the real proposition is to make `Type` and `Constraint` instance of a common `SORT`. So maybe

_Generalising over Type and Constraint_

(I would normally have made that a suggestion, but, probably because I'm in a train with flaky connection right now, Github won't let me)